data/moves.ts

```diff
@@ -15961,6 +15961,18 @@ export const Moves: {[moveid: string]: MoveData} = {
 				}
 				snatchUser.removeVolatile('snatch');
 				this.add('-activate', snatchUser, 'move: Snatch', '[of] ' + source);
+				// Call select onBeforeMove handlers
+				const handlersToRun = {"throatchop": 1, "healblock": 1};
+				const onBeforeMoveHandlers = this.findPokemonEventHandlers(snatchUser, "onBeforeMove")
+					.filter(handler => handlersToRun.hasOwnProperty(handler.effect.id));
+				const handlerResult = onBeforeMoveHandlers.reduce((result, handler) => {
+					if (handler.callback !== undefined) {
+						return handler.callback.bind(this)(snatchUser, null, move) && result;
```

Comment:
I don't know what the purpose of `findPokemonEventHandlers` is, so I don't know how best to refactor this to use arrow functions or avoid the use of `this`, but our [contributing guidelines](https://github.com/smogon/pokemon-showdown/blob/master/CONTRIBUTING.md#ES5-and-ES6) advise against using `Function#bind` for performance reasons.

Reply:
Probably the best way to implement this would be through a new flavor of ``Battle#runEvent()`` which passes a list of filtered ``effect.id``s to ``findPokemonEventHandlers``.

Reply:
Usually in cases like these we would just make a new event, right?

Reply:
Also

Reply:
Also, or simply hardcode the restrictions.